tests/checkers/rst-yamllint.py

```diff
@@ -0,0 +1,311 @@
+
+    def visit_system_message(self, node: nodes.system_message) -> None:
+        raise nodes.SkipNode
+
+    def visit_error(self, node: nodes.error) -> None:
+        raise nodes.SkipNode
+
+    def visit_literal_block(self, node: nodes.literal_block) -> None:
+        if "ansible-code-block" not in node.attributes:
+            if node.attributes["classes"]:
+                self.__results.append(
+                    {
+                        "path": self.__path,
+                        "line": node.line or "unknown",
+                        "col": 0,
+                        "message": f"Warning: found unknown literal block! Could be due to '::'. If not, please report this, this is likely a bug in the checker (could be an unsupported Sphinx directive). Node: {node!r}; attributes: {node.attributes}; content: {node.rawsource!r}",
+                    }
+                )
+            raise nodes.SkipNode
+
+        language = node.attributes["ansible-code-language"]
+        lineno = node.attributes["ansible-code-lineno"]
+
+        # Ok, we have to find both the row and the column offset for the actual code content
+        row_offset = lineno
+        found_empty_line = False
+        found_content_lines = False
+        content_lines = node.rawsource.count("\n") + 1
+        min_indent = None
+        for offset, line in enumerate(self.__content_lines[lineno:]):
+            stripped_line = line.strip()
+            if not stripped_line:
+                if not found_empty_line:
+                    row_offset = lineno + offset + 1
+                    found_empty_line = True
+            elif not found_content_lines:
+                found_content_lines = True
+                row_offset = lineno + offset
+
+            if found_content_lines and content_lines > 0:
+                if stripped_line:
+                    indent = len(line) - len(line.lstrip())
+                    if min_indent is None or min_indent > indent:
+                        min_indent = indent
+                content_lines -= 1
+            elif not content_lines:
+                break
+
+        min_source_indent = None
+        for line in node.rawsource.split("\n"):
+            stripped_line = line.lstrip()
+            if stripped_line:
+                indent = len(line) - len(line.lstrip())
+                if min_source_indent is None or min_source_indent > indent:
+                    min_source_indent = indent
+
+        col_offset = max(0, (min_indent or 0) - (min_source_indent or 0))
+
+        # Now that we have the offsets, we can actually do some processing...
+        if language not in {"YAML", "yaml", "yaml+jinja", "YAML+Jinja"}:
+            if language is None:
+                self.__results.append(
+                    {
+                        "path": self.__path,
+                        "line": row_offset + 1,
+                        "col": col_offset + 1,
+                        "message": "Literal block without language!",
```

Comment:
```suggestion
                        "message": "Literal blocks must specify a language!",
```

Reply:
Should we provide some language examples?

Reply:
Implemented in 1196a3196f1fda1dd3d2cfaf8ce589845e764198.

Reply:
Thanks @felixfontein